satpy/tests/reader_tests/test_eum_l2_grib.py

```diff
@@ -133,12 +133,16 @@ def test_seviri_data_reading(da_, xr_, setup_reader):
 
             dataset_id = make_dataid(name="dummmy", resolution=3000)
 
+            # Check that end_time is None for SEVIRI before the dataset has been loaded
+            assert reader.end_time is None
+
             common_checks(ec_, reader, mock_file, dataset_id)
 
-            # Check end_time
+            # Check that end_time is now a valid datetime.datetime object after the dataset has been loaded
             assert reader.end_time == datetime.datetime(year=2020, month=10, day=20,
                                                         hour=19, minute=50, second=0)
 
+
```

Comment:
Could this test case be split into several specific tests?

Reply:
hm, I've been trying to refactor this into something better, but so far with no satisfying results. I have previously created an issue to work on removing the mocking #2932, do you think we could move this test optimization to the PR that will deal with the mocking (since it will anyway lead to quite some refactoring)?

Reply:
Yeah, sounds good! Feel free to merge.

Reply:
Thanks! Nah, you do it, you're the reviewer ;)